src/pages/settings/Settings.tsx

```diff
@@ -190,6 +250,159 @@ const Settings: FC = () => {
       };
     });
 
+  userDefinedConfigs.forEach((_, index) => {
+    rows.push({
+      key: `new-config-${index}`,
+      columns: [
+        {
+          content: false,
+          role: "rowheader",
+          className: "group",
+          "aria-label": "Group",
+        },
+        {
+          content: (
+            <>
+              <Input
+                aria-label={`new-user-defined-key-${index}`}
+                id={`new-user-defined-key-${index}`}
+                placeholder="User key"
+                type="text"
+                value={userDefinedConfigs[index].key}
+                error={
+                  isKeyDuplicate(userDefinedConfigs[index].key) && (
+                    <p>Setting with this name already exists</p>
+                  )
+                }
+                onChange={(e) => {
+                  setUserDefinedConfigs((prev) => {
+                    const copy = [...prev];
+                    copy[index] = {
+                      ...copy[index],
+                      key: e.target.value,
+                    };
+                    return copy;
+                  });
+                }}
+              />
+              {index == userDefinedConfigs.length - 1 && (
+                <p className="p-text--small u-text--muted u-no-margin--bottom">
+                  <ConfigFieldDescription
+                    description={
+                      "Key will be saved as user.`{your-key}`. Enter only the part after user."
+                    }
+                  />
+                </p>
```

Comment:
The spacing here is off and this is a bit too complicated. You can set the help property on `Input` instead to keep this simple.